### PR TITLE
Add flowLockErrorMessage option to pass customized error message when flow is locked due to calling setFlowLock API

### PR DESCRIPTION
**Pull request by funcheetah**
This PR adds flowLockErrorMessage option in ProjectManagerServlet which allows user to pass customized error message to show in flow page when flow is locked due to calling setFlowLock API.

### CODE REVIEW

File: azkaban-common/src/main/java/azkaban/executor/ExecutableFlow.java [hunk 1]
```diff
@@ -218,6 +222,14 @@ public void setAzkabanFlowVersion(final double azkabanFlowVersion) {
 
   public void setLocked(boolean locked) { this.isLocked = locked; }
 
+  public String getFlowLockErrorMessage() {
+    return this.flowLockErrorMessage;
+  }
+
+  public void setFlowLockErrorMessage(String flowLockErrorMessage) {
```
Comment (djaiswal83) [inline azkaban-common/src/main/java/azkaban/executor/ExecutableFlow.java]: final String flowLockErrorMessage.

File: azkaban-common/src/main/java/azkaban/flow/Flow.java [hunk 2]
```diff
@@ -470,4 +477,12 @@ public void setProjectId(final int projectId) {
   public boolean isLocked() { return this.isLocked; }
 
   public void setLocked(boolean locked) { this.isLocked = locked; }
+
+  public String getFlowLockErrorMessage() {
+    return this.flowLockErrorMessage;
+  }
+
+  public void setFlowLockErrorMessage(String flowLockErrorMessage) {
```
Comment (djaiswal83) [inline azkaban-common/src/main/java/azkaban/flow/Flow.java]: Same, use final

File: azkaban-web-server/src/main/java/azkaban/webapp/servlet/ProjectManagerServlet.java [hunk 3]
```diff
@@ -1620,7 +1631,8 @@ private void handleFlowPage(final HttpServletRequest req, final HttpServletRespo
         page.add("isLocked", flow.isLocked());
         if (flow.isLocked()) {
           final Props props = this.projectManager.getProps();
-          final String lockedFlowMsg = String.format(props.getString(ConfigurationKeys
+          final String flowLockErrorMessage = flow.getFlowLockErrorMessage();
+          final String lockedFlowMsg = flowLockErrorMessage != null ? flowLockErrorMessage : String.format(props.getString(ConfigurationKeys
```
Comment (djaiswal83) [inline azkaban-web-server/src/main/java/azkaban/webapp/servlet/ProjectManagerServlet.java]: indentation

File: azkaban-common/src/main/java/azkaban/executor/ExecutableFlow.java (hunk 1 above)
Comment (funcheetah) [reply azkaban-common/src/main/java/azkaban/executor/ExecutableFlow.java]: Thanks for the comment. Done.

File: azkaban-web-server/src/main/java/azkaban/webapp/servlet/ProjectManagerServlet.java (hunk 3 above)
Comment (funcheetah) [reply azkaban-web-server/src/main/java/azkaban/webapp/servlet/ProjectManagerServlet.java]: Thanks for the comment. Done.

File: azkaban-common/src/main/java/azkaban/flow/Flow.java (hunk 2 above)
Comment (funcheetah) [reply azkaban-common/src/main/java/azkaban/flow/Flow.java]: Thanks for the comment. Done.

**Pull request merged**